Library/Homebrew/version.rb

```diff
@@ -379,6 +379,21 @@ def self._parse(spec, detected_from_url:)
   end
   private_class_method :_parse
 
+  DOT_OPTIONAL = /(?:\d+(?:\.\d+)*)/.source.freeze
+  private_constant :DOT_OPTIONAL
+
+  DOT_REQUIRED = /(?:\d+(?:\.\d+)+)/.source.freeze
+  private_constant :DOT_REQUIRED
+
+  MINOR_OR_PATCH = /(?:\d+(?:\.\d+){1,2})/.source.freeze
+  private_constant :MINOR_OR_PATCH
+
+  BIN_SUFFIX = /(?:[._-](?i:bin|dist|stable|src|sources?|final|full))/.source.freeze
+  private_constant :BIN_SUFFIX
```

Comment:
```suggestion
  COMMON_SUFFIXES = /(?:[._-](?i:bin|dist|stable|src|sources?|final|full))/.source.freeze
  private_constant :COMMON_SUFFIXES
```

or `DISCARDABLE_SUFFIXES` or `IGNORABLE_SUFFIXES` or something

Reply:
Maybe `CONTENT_SUFFIXES`, since they indicate something about the contents of the archive file?

Reply:
I think `CONTENT_SUFFIX` is good; this regex matches a single instance of `-bin`, `-dist`, `-stable`, etc., so the singular form "suffix" makes the most sense